### DISCUSSION TITLE
Checking for NOASSERTION in license policy?

### DISCUSSION THREAD

**Discussion by vs-sfalsig** [DependencyTrack/dependency-track#2810] [Q&A]
Hi,

I'm somewhat new to DependencyTrack. Is there any way of creating a license policy to check for the license ID being 'NOASSERTION'? Looks like I can only select form the existing policy list, which doesn't include this (and can't see a way to add it).

For reference, my usecase is that I want to use dependency-track for checking the license of various Python libraries, based on SBOMs generated with the cyclonedx-py tool. However, (from what I understand) due to Python packages not specifying their license in SPDX format, the generated SBOM license information can't really be used by dependency-track, as it just tracks the 'name' field, which isn't much use in automated analysis ('BSD-3-Clause' / 'BSD 3-Clause' / '3-Clause BSD License' / 'https://opensource.org/licenses/BSD-3-Clause' / etc. / etc.).
I'm thus setting up a small tool for translating this internally and adding the license *id* to the SBOM, outputting 'NOASSERTION' as the id, if it couldn't determine the actual license (based on https://spdx.github.io/spdx-spec/v2.3/package-information/#713-concluded-license-field). I want DependencyTrack to warn on this though, so that we can easily track cases that need further manual intervention.

Would appreciate any input on whether this is possible! - or whether I'm just understanding this the wrong way around...;)

Currently on DependencyTrack v4.3.6, so apologies if this is a non-issue in the newest version. I'm looking at getting our instance upgraded, but not immediately possible.

Thanks in advance!

**Comment by stevespringett** [comment]
NOASSERTION is not a valid SPDX license identifier and would violate the CycloneDX spec if used as an identifier. SPDX uses "NOASSERTION" because the license is required and this is their way of getting around that. In CycloneDX, its just the opposite. If a component does not have a license, then it hasn't been asserted. There's no need for NOASSERTION. 

I'm not sure if creating a policy for a component without a license is possible or not, but I would try that. If not possible, that would be a good enhancement. Also,. v4.3.6 is really old at this point, so please consider upgrading.

**Comment by nscuro** [comment]
Matching of *unresolved* licenses in policies is possible since 4.7.0: https://github.com/DependencyTrack/dependency-track/pull/2214

As Steve pointed out, instead of using `NOASSERTION` in your tool, simply omit the license ID entirely. You can then use a `License IS "unresolved"` policy in DT to alert on occurrences of them.

**Comment by vs-sfalsig** [comment]
Hi Steve, Niklas,

Great - thank you for the quick responses! I'll try that once I get my instance upgraded, and report back ;)
